slides/06-Contributing/02-testing.md

```diff
@@ -2,14 +2,292 @@
+
+These tests are specified directly in a GitHub Action workflow: [.github/workflows/compilers-tests.yaml](https://github.com/gem5/gem5/blob/v24.0/.github/workflows/compilers-tests.yaml)
+
+These tests use a series of docker images to test compilation of various gem5 conifgurations with different compilers.
+
+---
+
+## TestLib Tests
+
+TestLib tests are the most important tests in gem5.
+These tests run gem5 simulations and verify the output of the simulation.
+The tests are written in Python and use the "testlib" framework to run the simulations and verify the output.
+
+The tests are run using the `./main.py` command in the "test" directory of the gem5 repository.
+
+It's useful to just focus on a subdirectory of tests when running tests:
+
+```shell
+./main.py run gem5/memory
+```
+
+The above will only run the "quick" tests in the "tests/gem5/memory" directory.
+The "quick" tests are the testlib tests run in the CI pipline. To run the tests in the "daily" or "weekly" test suites, you can use the `--length` to specify `long` or `very-long` (`quick` is the `length` default).
+
+---
+
+The `./main.py list` command can be used to list all the tests in a directory, which we'll demonstrate here:
+
+
+```shell
+# List all the long tests in tests/gem5/memory: Those run in the Daily Tests.
+./main.py list --length long gem5/memory
+
+# lists all the very long tests in tests/gem5/memory: Those run in the Weekly Tests.
+./main.py lists --length very-long gem5/memory
+```
+
+---
+
+### How TestLib Tests are declared
+
+Let's look at ["tests/gem5/m5_util"](https://github.com/gem5/gem5/blob/v24.0/tests/gem5/m5_util) to see how a test is declared.
+
+In this directory there is "test_exit.py".
+Any file with the prefix "test_" is considered a test by the testlib framework and will be run when the tests are executed.
+
+"configs" is a directory of configuration scripts that are used to run the tests defined in "test_exit.py".
+
+Now, let's look into "test_exit.py" and see how the tests are declared.
+
+---
+
+```python
+import re
+
+# Import the testlib framework. This is required.
+from testlib import *
+```
+---
+
+```py
+# Here we define a regular expression to match the output of the simulation.
+
+m5_exit_regex = re.compile(
+    r"Exiting @ tick \d* because m5_exit instruction encountered"
+)
+
+# ...
+
+# Here we define the verifier using the regular expression.
+a = verifier.MatchRegex(m5_exit_regex)
+```
+
+---
+
+```py
+gem5_verify_config(
+    name="m5_exit_test", # The test name
+    verifiers=[a], # The verifier (must be iterable!)
+    fixtures=(),
+    config=joinpath( # The path to the config file to run.
+        config.base_dir,
+        "tests",
+        "gem5",
+        "m5_util",
+        "configs",
+        "simple_binary_run.py",
+    ),
+    config_args=[ # The arguments to pass to the config file.
+        "x86-m5-exit",
+        "--resource-directory",
+        resource_path,
+    ],
```

Comment:
Missing `)`, I think. Something is wrong with the syntax here.

Reply:
Fixed